### PR TITLE
Add simplified YodaConditionsFixer

### PR DESCRIPTION
**Pull request by keradus**
See:
- #700 for temporary remove YodaConditionsFixer
- #693 for discuss what need to be changed in fixer and discussion
- #698 for some little fixes

### CODE REVIEW

File: Symfony/CS/Tests/Fixer/Symfony/YodaConditionsFixerTest.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        $this->makeTest($expected, $input);
+    }
+
+    public function provideExamples()
+    {
+        return array(
+            // valid conditions that should not be changed
+            array('<?php return array(2) == $a;'),
+            array('<?php return $a == array(2);'),
```
Comment (localheinz) [inline Symfony/CS/Tests/Fixer/Symfony/YodaConditionsFixerTest.php]: Shouldn't this actually have to be changed? The left-hand side is a variable while the right-hand side is not.

File: Symfony/CS/Tests/Fixer/Symfony/YodaConditionsFixerTest.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            array('<?php return array($a) == $a;'),
+            array('<?php return $a == array($a);'),
+            array('<?php return $this->getStuff() === $myVarirable;'),
+            array('<?php return $myVarirable === $this->getStuff();'),
```
Comment (localheinz) [inline Symfony/CS/Tests/Fixer/Symfony/YodaConditionsFixerTest.php]: Doesn't make a difference for the test, but probably should be `$myVariable`, no?

**Comment by localheinz** [comment]
@keradus 

Would love to see this fixer finished!

File: Symfony/CS/Tests/Fixer/Symfony/YodaConditionsFixerTest.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            array('<?php return 2 * $myVar % 3 === $a;'),
+            array('<?php return $a === 2 * $myVar % 3;'),
+            array('<?php return ($a & self::MY_BITMASK) === $a;'),
+            array('<?php return $a === ($a & self::MY_BITMASK);'),
+            array('<?php return count($this->array[$var]) === $a;'),
+            array('<?php return $a === count($this->array[$var]);'),
```
Comment (localheinz) [inline Symfony/CS/Tests/Fixer/Symfony/YodaConditionsFixerTest.php]: I think this condition would need to be changed by the fixer, too, right?

File: Symfony/CS/Tests/Fixer/Symfony/YodaConditionsFixerTest.php (hunk 1 above)
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/YodaConditionsFixerTest.php]: Hi!
This should be 2 fixers. What you want will be sth like ExtendedYodaFixer, what we have some time ago. That fixer won't be applied on Symfony. Please look at commit history of this PR, the first one is a revert of removing a fixer.
The second one will not change this and will be sth like SimplifiedYodaFixer. This fixer will be applied on Symfony.

File: Symfony/CS/Fixer/Symfony/YodaConditionsFixer.php [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+     * Checks whether the tokens between the given start and end describe a
+     * variable.
+     *
+     * @param Tokens $tokens The token list
+     * @param int    $start  The first index of the possible variable
+     * @param int    $end    The last index of the possible varaible
+     *
+     * @return bool Whether the tokens describe a variable
+     */
+    private function isVariable(Tokens $tokens, $start, $end)
```
Comment (gharlan) [inline Symfony/CS/Fixer/Symfony/YodaConditionsFixer.php]: missing variable variables `$$var`. They are tokenized as `$ + T_VARIABLE`.

File: Symfony/CS/Fixer/Symfony/YodaConditionsFixer.php [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+        return $startLeft;
+    }
+
+    /**
+     * Checks whether the tokens between the given start and end describe a
+     * variable.
+     *
+     * @param Tokens $tokens The token list
+     * @param int    $start  The first index of the possible variable
+     * @param int    $end    The last index of the possible varaible
```
Comment (gharlan) [inline Symfony/CS/Fixer/Symfony/YodaConditionsFixer.php]: typo "varaible"

**Comment by keradus** [comment]
Please hold with review @gharlan.
This PR is frozen.

**Comment by gharlan** [comment]
ok :)

**Comment by keradus** [comment]
Closed by merge of #2446

**Pull request closed without merging**